Upgrade packages

dslucas opened the pull request:
We also make a few minor code changes corresponding to changes in the dependencies.

dslucas committed:
Upgrade packages

We also make a few minor code changes corresponding to changes in the dependencies.

rachelrosengoogle commented on package.json:
```diff
@@ -13,44 +13,44 @@
     "test": "mocha build/server/serving_test.js"
   },
   "dependencies": {
```
Could you update the package version? (For some reason github won't let me leave a comment on that exact line, but I mean the version a few lines up). We'll still have to publish to npm after this as a separate step.

dslucas replied:
Done.

rachelrosengoogle commented on package.json:
```diff
@@ -13,44 +13,44 @@
+    "supertest": "^4.0.2",
+    "tslint": "^5.20.0",
+    "typedoc": "^0.15.0",
+    "typescript": "^3.6.3",
```
This is a version of typescript that is not yet supported by Angular (see https://github.com/angular/angular/issues/32380) which means that we would not be able to update the projects that use this server (perspectiveapi-authorship-demo and the website) to be able to use this.

Could you make this at most typescript 3.5? We'll still probably have to upgrade the packages that depend on this in order to make use of these changes, since Typescript versions for Angular projects are tied to what Angular version being used.

I think it's fine to update this repo independently because of npm versioning, as long as we're aware of these potential issues when we update the version of this server that we're dependent on in those projects.

dslucas replied:
Thanks for catching that. I've limited the version to <=3.5.

dslucas committed:
Update packge version to v0.1.6

We also limit the version of TypeScript to maintain compatibility with Angular.

rachelrosengoogle commented on package.json:
```diff
@@ -13,44 +13,44 @@
+    "supertest": "^4.0.2",
+    "tslint": "^5.20.0",
+    "typedoc": "^0.15.0",
+    "typescript": "<=3.5",
```
I think you can just specify something like ^3.5.0 or the equivalent, which is a narrower range than this and better for build dependencies. Other libraries you're changing here are linked to specific typescript versions so it's better to define the scope as narrowly as possible.

dslucas replied:
Ah, makes sense. I specified to ^3.5.3, which was the latest version before 3.6.

dslucas committed:
Restrict the TypeScript version to 3.5.3
